base/setup/lib/utils/filesup.c

```diff
@@ -71,78 +68,84 @@ SetupCreateSingleDirectory(
                           NULL,
                           0);
     if (NT_SUCCESS(Status))
-    {
         NtClose(DirectoryHandle);
-    }
-
-    RtlFreeUnicodeString(&PathName);
 
     return Status;
 }
 
 NTSTATUS
 SetupCreateDirectory(
-    IN PCWSTR PathName)
+    _In_ PCWSTR DirectoryName)
 {
     NTSTATUS Status = STATUS_SUCCESS;
-    PWCHAR PathBuffer = NULL;
+    UNICODE_STRING PathNameU;
+    PWCHAR PathBuffer;
     PWCHAR Ptr, EndPtr;
-    ULONG BackslashCount;
-    ULONG Size;
-
-    Size = (wcslen(PathName) + 1) * sizeof(WCHAR);
-    PathBuffer = RtlAllocateHeap(RtlGetProcessHeap(), HEAP_ZERO_MEMORY, Size);
-    if (PathBuffer == NULL)
-        return STATUS_INSUFFICIENT_RESOURCES;
-
-    wcscpy(PathBuffer, PathName);
-    EndPtr = PathBuffer + wcslen(PathName);
 
-    Ptr = PathBuffer;
+    if (!RtlCreateUnicodeString(&PathNameU, DirectoryName))
+        return STATUS_NO_MEMORY; // STATUS_INSUFFICIENT_RESOURCES;
```

Comment:
```suggestion
        return STATUS_NO_MEMORY;
```
you can't decide yet which one of the two to return here? And therefore you think maybe it's better to keep the old thing lingering around in the code, in case someone else will decide to flip it back later? That's not a good reason to keep it. Strip dead code **today**. We do have a version control system for code archeology!

Reply:
Addressed as part of improvements done in PR 7257